test/functional/tests/volumes/test_many_lvms_on_single_core.py

```diff
@@ -0,0 +1,114 @@
+          1 cache, 1 core, 16 lvms (1 PV, 1 VG, 16 LV).
+        pass_criteria:
+          - CAS devices created successfully.
+          - LVMs created successfully.
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Create CAS device."):
+
+        cache_dev, core_dev = prepare_devices()
+
+        cache = casadm.start_cache(cache_dev[0], force=True)
+        core = cache.add_core(core_dev[0])
+
+    with TestRun.step("Create LVMs on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 1
+        vg_num = 1
+        lv_num = 16
+        cache_num = 1
+        cas_dev_num = 1
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core, config)
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Flush buffers"):
+        for lvm in lvms:
+            TestRun.executor.run(f"hdparm -f {lvm.path}")
+
+    with TestRun.step("Create init config from running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_before_reboot = fs_utils.read_file(opencas_conf_path)
+
+    with TestRun.step("Reboot system."):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Validate running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_after_reboot = fs_utils.read_file(opencas_conf_path)
+
+        if config_after_reboot == config_before_reboot:
+            TestRun.LOGGER.info(f"Configuration is as expected")
+        else:
+            TestRun.LOGGER.error(f"Configuration changed after reboot.")
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Remove LVMs and clean out config modification."):
+        Lvm.remove_all()
+        Lvm.remove_global_filter_from_config()
+
+
+def prepare_devices():
+    cache_device = TestRun.disks['cache']
+    core_device = TestRun.disks['core']
+
+    cache_device.create_partitions([Size(8, Unit.GibiByte)])
+    core_device.create_partitions([Size(8, Unit.GibiByte)])
+
+    return cache_device.partitions, core_device.partitions
+
+
+def run_fio_on_lvm(volumes):
+    fio_run = (Fio().create_command()
+               .read_write(ReadWrite.randrw)
+               .io_engine(IoEngine.sync)
+               .io_depth(1)
+               .time_based()
+               .run_time(datetime.timedelta(seconds=180))
+               .do_verify()
+               .verify(VerifyMethod.md5)
+               .block_size(Size(1, Unit.Blocks4096)))
+
+    for lvm in volumes:
+        fio_job = fio_run.add_job()
+        fio_job.target(lvm)
+        fio_job.size(lvm.size)
```

Comment:
I am not sure if it will work correctly while it is set on some new variable called "fio_job".
I'd propose using construction like:
fio_run.add_job().target(lvm).size(lvm.size)